### PR TITLE
CLASSIFICATION: Improving the metrics computation.

### PR DESCRIPTION
**Pull request by oualib**
- It is now possible to compute multiclass classifier micro/macro/weighted metrics.
 - regression and classification reports are more flexible and allow to compute more metrics and list of metrics.
 - many new metrics for classification and regression.

MISC:
 - code formatting
 - correcting wrong types.

@VerticaDataScientist / @mat-shyR can you please review the following files:
 - classification
 - regression
 - base of machine_learning/vertica

@mail4umar 
 - Please review the entire PR I've probably missed some things. I need to complete it to move to the next one.

### CODE REVIEW

File: verticapy/machine_learning/metrics/classification.py [hunk 1]
```diff
@@ -288,31 +308,58 @@ def critical_success_index(
...
-    csi = tp / (tp + fn + fp) if (tp + fn + fp != 0) else 0
-    return csi
+    return _compute_final_score(
+        _accuracy_score,
+        y_true,
+        y_score,
+        input_relation,
+        average=average,
+        labels=labels,
+        pos_label=pos_label,
```
Comment (mail4umar) [inline verticapy/machine_learning/metrics/classification.py:344]: Instead i propose we can use the following **locals() argument to make it even more cleaner:

def sub_fun(fun_call,one_param,two_param,three_param):
  if fun_call==1:
    return one_param+two_param+three_param

def main_fun(one_param,two_param,three_param):
  return sub_fun(1,**locals())

main_fun(1,2,3)

File: verticapy/machine_learning/vertica/base.py [hunk 2]
```diff
@@ -1718,41 +1774,41 @@ class = '{{}}',
...
         Returns the input which represents the  model's 
         probabilities.
         """
         return self.deploySQL(allSQL=True)[0].format(pos_label)
 
     def _get_y_score(
-        self, pos_label: PythonScalar = None, cutoff: PythonNumber = 0.5,
+        self,
+        pos_label: Optional[PythonScalar] = None,
+        cutoff: Optional[PythonNumber] = None,
```
Comment (mail4umar) [inline verticapy/machine_learning/vertica/base.py:1794]: pythonnumber = None? 
double check

**Review by mail4umar — APPROVED** [review]

File: verticapy/machine_learning/metrics/classification.py [hunk 3]
```diff
@@ -288,31 +308,58 @@ def critical_success_index(
...
     """
-    tn, fn, fp, tp = _compute_tn_fn_fp_tp(y_true, y_score, input_relation, pos_label)
-    csi = tp / (tp + fn + fp) if (tp + fn + fp != 0) else 0
-    return csi
+    return _compute_final_score(
+        _accuracy_score,
+        y_true,
+        y_score,
+        input_relation,
+        average=average,
```
[the remark this answers is not in the document]
Comment (oualib) [reply verticapy/machine_learning/metrics/classification.py]: will do it in the future. Thanks a lot for the tip.

File: verticapy/machine_learning/metrics/classification.py (hunk 1 above)
Comment (oualib) [reply verticapy/machine_learning/metrics/classification.py]: will do it in the future. Thanks a lot for the tip.

**Pull request merged**